src/main/java/seedu/address/model/achievement/AchievementRecord.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.achievement;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import javafx.beans.property.SimpleObjectProperty;
+
+/**
+ * Represents a record of the user's achievements while using the task manager.
+ * Achievements include the experience points(xp) earned by completing the task and the level a user has reached.
+ * Guarantees: details are present and not null.
+ */
+public class AchievementRecord {
+
+    private Xp xp;
+    private Level level;
+
+
+    /**
+     * Constructs a {@code AchievementRecord}.
+     * Xp value is initialized to 0 and Level initialized to LEVEL_1.
+     */
+    public AchievementRecord() {
+        this.xp = new Xp();
+        this.level = Level.LEVEL_1;
+    }
+
+    /**
+     * Constructs a {@code AchievementRecord}.
+     * Both fields must be present.
+     */
+    public AchievementRecord(Xp xp, Level level) {
+        requireAllNonNull(xp, level);
+        this.xp = xp;
+        this.level = level;
+    }
+
+    public Level getLevel() {
+        return level;
+    }
+
+    public void setLevel(Level level) {
+        this.level = level;
+    }
+
+    public Xp getXp() {
+        return xp;
+    }
+
+    public void setXp(Xp xp) {
+        this.xp = xp;
+    }
+
+    /**
+     * Resets the existing data of this {@code AchievementRecord} with {@code newData}.
+     */
+    public void resetData(AchievementRecord newData) {
+        requireNonNull(newData);
+
+        setXp(newData.getXp());
+        setLevel(newData.getLevel());
+    }
+
+    /**
+     * Updates the Xp field of this {@code AchievementRecord} with new xp value.
+     * Triggers the update of Level with xp.
+     */
+    public void updateXp(Integer xp) {
```

Comment:
This method would be more informative if it's named addXP

Reply:
I named it updateXp because it might be used to decrease xp as well, for overdue tasks or when user change the status of a task from complete to in progress.